apollo-router/src/services/subgraph_service.rs

```diff
@@ -103,154 +109,212 @@ impl tower::Service<crate::SubgraphRequest> for SubgraphService {
     }
 
     fn call(&mut self, request: crate::SubgraphRequest) -> Self::Future {
-        let crate::SubgraphRequest {
-            subgraph_request,
-            context,
-            ..
-        } = request;
-
         let clone = self.client.clone();
-
-        let mut client = std::mem::replace(&mut self.client, clone);
+        let client = std::mem::replace(&mut self.client, clone);
         let service_name = (*self.service).to_owned();
 
-        Box::pin(async move {
-            let (parts, body) = subgraph_request.into_parts();
-            let body = serde_json::to_string(&body).expect("JSON serialization should not fail");
+        let make_calls = async move {
+            // Try the subgraph_request (body) without query first
```

Comment:
will this cause a race condition for cold start? can we explore a slow start mechanism to avoid this race condition. it might just be documentation (not sure).